aruha-738 parameterize data directories

rcillo opened the pull request:
In order to leverage high performance IO disks from AWS i3 instance family,
it's necessary to mount a volume on a given path and later write data
files to this volume.

This commit exposes two environmnent variables that can be used to
specify the path to the mounted volume.

Commit:
aruha-738 parameterize data directories

In order to leverage high performance IO from AWS i3 instance family,
it's necessary to mount a volume on a given path and later write data
files to this volume.

This commit exposes two environmnent variables that can be used to
specify the path to the mounted volume.

CyberDem0n commented on exhibitor-appliance.yaml:
```diff
@@ -19,6 +19,12 @@ SenzaInfo:
         Default: ""
+    - CustomTransactionsDirectory:
+        Description: "Use custom transactions directory"
+        Default: "/opt/exhibitor/transactions"
```
If the `CustomTransactionsDirectory` and `CustomSnapshotsDirectory` are not specified in command line it will default to `/opt/exhibitor/transactions` and `/opt/exhibitor/snapshots` which are probably not writeable.
That means a new docker image wont work with this senza template.

Also be careful with i3 instances. AWS documentation tells that Ubuntu 14.04 (on which taupage is based) has outdated support of NVMe. New taupage will include a kernel from 16.04. But it's not released yet publicly.

rcillo replied:
@CyberDem0n I added some complementary directory creation commands to the `run.sh` file as well, so to be sure that these directories are going to be created with the right permissions. Lines [17](https://github.com/zalando-incubator/exhibitor-appliance/pull/31/files#diff-1b0c2b516b83393edb7200ad5ff12181R17) and [21](https://github.com/zalando-incubator/exhibitor-appliance/pull/31/files#diff-1b0c2b516b83393edb7200ad5ff12181R21).

I'm not sure if this is the best approach though.

CyberDem0n replied:
If directory already exists it wont change permissions.

Best approach...
Actually It looks fine, just do a small change `/opt/exhibitor/ => /opt/zookeeper/` to be sure that nothing will break with updated senza template.

Commit:
Avoid accidentally changing the directory

It used to be stored under zookeeper directory and not exhibitor
directory. So we should not change the defaults.

CyberDem0n commented:
:+1:

rcillo commented:
👍

rcillo commented:
@CyberDem0n do you think this is good enough to merge and publish a new oficial image?

ikitiki commented:
👍
